packages/framework/src/workflow.ts

```diff
@@ -27,9 +31,51 @@ export function workflow<
   workflowId: string,
   execute: Execute<T_Payload, T_Control>,
   workflowOptions?: WorkflowOptions<T_PayloadSchema, T_ControlSchema>
-): DiscoverWorkflowOutput {
+): Workflow<T_Payload> {
   const options = workflowOptions ? workflowOptions : {};
 
+  const apiClient = initApiClient(process.env.NOVU_SECRET_KEY as string);
```

Comment:
We are using the API client here in quite a dumb way.

I'm not particularly happy about this for a couple reasons:
1. Each workflow will have a client instantiated
2. It doesn't use the `secretKey` provided via the client options

Would love to hear any suggestions for improvement here.

Reply:
Additionally, I chose not to use the `@novu/node` package right now and use `fetch` directly instead, because using the former felt like quite a sledgehammer for what we need right now.

When we look at consolidating the `@novu/framework` and `@novu/node` packages into a single package, we should revisit this.

Reply:
I agree 100%. 

1. As suggested, I'd like to use a basic fetch at this point. 
2. I like the factory pattern for this kind of this such as function `buildAPIClient({ apiKey, ...options })`. You can call that once and assign it to the current framework `Client` object as a private prop.

Reply:
The `workflow()` doesn't have access to the `Client.apiClient` as the Workflows come before the Client, not after. The current approach of instantiating a global client here isn't ideal, but it saves us from having a global client living in the installed NPM package.

@SokratisVidros , do we want to move ahead with this for now and revisit as part of adding the `@novu/api` REST-API client?

Reply:
Sure.